terraform/cloud_function/scheduled/main.tf

```diff
@@ -0,0 +1,69 @@
+/*
+Cloud Function invoked by a scheduled job through a HTTP request
+
+Often used to do things that need to be executed at regular intervals,
+like pulling data from external sources or doing aggregations
+*/
+
+resource "random_string" "random" {
+  length           = 16
+  special          = true
+  override_special = "/@£$"
+}
+
+resource "google_storage_bucket_object" "functioncode" {
+  name   = format("http_function_sources/%s/sourcecode%s.zip", var.function_name, random_string.random.result)
+  bucket = var.source_code_bucket_name
+  source = "${var.source_code_root_path}/${var.function_name}/${var.function_name}.zip"
+}
+
+resource "google_cloudfunctions_function" "function" {
+  available_memory_mb           = var.function_memory
+  entry_point                   = var.function_entry_point
+  environment_variables         = var.function_env_vars
+  name                          = format("%s%s", var.function_name, var.branch_suffix)
+  project                       = var.project_id
+  region                        = var.project_region
+  runtime                       = var.function_runtime
+  service_account_email         = var.function_service_account_email
+  source_archive_bucket         = var.source_code_bucket_name
+  source_archive_object         = google_storage_bucket_object.functioncode.name
+  timeout                       = var.function_timeout
+  trigger_http                  = true
+  vpc_connector                 = var.vpc_connector
+  vpc_connector_egress_settings = var.vpc_connector_egress_settings
+}
+
+# Add cloud scheduler job
+resource "google_cloud_scheduler_job" "casco_listing_job" {
+  count    = var.branch_suffix == "" ? 1 : 0 # Only enable job on production to avoid branches eating each others lunch
```

Comment:
Having this logic here is very ugly. We can make this a variable, `var.enable_scheduler` or something, and leave the logic of when to enable in the place where the module is used

Reply:
agree 👍